### PR TITLE
Fixes #16224: Missing documentation on openssl incompatibilities between 4.x and 5.0

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/16224

### CODE REVIEW

File: src/reference/modules/installation/pages/upgrade.adoc [hunk 1]
```diff
@@ -74,6 +74,58 @@ This has been fixed in the packages, but the problem is caused by the
...
+Yet Rudder 4.1, 4.2 or 4.3 use OpenSSL provided by the system and some Linux
+distributions supported by these versions of Rudder only provide flavors of 
+OpenSSL 1.0.1. 
+
+Solution:
+
+As far as we know, there is three solutions available to migrate Rudder
+servers and agent from 4.x to 5.0 and avoid the communication problem:
+
+- the simplest solution is to contact Rudder support which can provide a specialiy
```
Comment (amousset) [inline src/reference/modules/installation/pages/upgrade.adoc]: specially

File: src/reference/modules/installation/pages/upgrade.adoc [hunk 2]
```diff
@@ -74,6 +74,58 @@ This has been fixed in the packages, but the problem is caused by the
...
+compiled version of Rudder policy server with OpenSSL 1.0.2 for use during the
+migration. Once the migration is done, the common Rudder policy server component
+can be reinstalled to ensure using the last version of OpenSSL. Rudder support can
+also provide compatible scale out relays to segment your migration.
```
Comment (amousset) [inline src/reference/modules/installation/pages/upgrade.adoc]: nothing specific for relays

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**